src/actions/sso/authenticationRequest.ts

```diff
@@ -21,10 +23,23 @@ export const consumeAuthenticationRequest = (
   authenticationRequest: JSONWebToken<Authentication>,
   isDeepLinkInteraction: boolean = false,
 ): ThunkAction => async (dispatch, getState, backendMiddleware) => {
-  const { identityWallet } = backendMiddleware
+  const { identityWallet, registry } = backendMiddleware
   await identityWallet.validateJWT(authenticationRequest)
+  const { did: requesterDid, publicProfile } = await registry.resolve(
+    keyIdToDid(authenticationRequest.issuer)
+  )
+
+  const parsedProfile = publicProfile
+    ? omit(['id'], publicProfile.toJSON().claim)
+    : {}
+
+  const requesterSummary = mergeRight(
```

Comment:
This merge is not needed, there are no overlapping fields, so this is equivalent to just
```ts
 const requesterSummary = {
  did: requesterDid,
  publicProfile: parsedProfile
}
```

Reply:
This was mentioned in the other review as well, i.e. [here](https://github.com/jolocom/smartwallet-app/pull/1456#pullrequestreview-286746716).

Both this and the aforementioned PR simply duplicate the functionality already present in `consumeCredentialRequest` (which was added as part of the BDR integration).

Eventually this logic can be abstracted in a separate function (as per the [issue](https://github.com/jolocom/smartwallet-app/issues/1459)).

The only reason `mergeRight` was originally used here was to avoid typescript errors :frowning_face: (partially due to time restrictions), because `SignedCredential.claim` contains a type which does not match type for `IdentitySummary`.

This can be addressed with better typings, or a ts-ignore basically.

Reply:
Was not aware of the previous discussion! Thanks!